raiden/network/transport/matrix/utils.py

```diff
@@ -392,6 +392,37 @@ def join_broadcast_room(client: GMatrixClient, name: str, servers: Sequence[str]
     return broadcast_room
 
 
+def register(client: GMatrixClient, signer: Signer, base_username: str) -> User:
+    """ Register a new random userid with the chosen Matrix server. """
+    server_url = client.api.base_url
+    server_name = urlparse(server_url).netloc
+
+    # A deterministic userid cannot be used since that would be allow for an
+    # DoS attack, were an attacker registers the userid before the real user.
+    # To fix this a random number is added to the username.
+    username = f"{base_username}.{Random().randint(0, 0xffffffff):08x}"
```

Comment:
Can we keep at least the default (`0`) case? I think there's value in having the default userId without the suffix, and adding it only if login/registration fails

Reply:
Why? Doesn't that means we have two cases to test? And all our tests will only try the happy path without the suffix?

Reply:
The tests doesn't need to test a specific case between both, just check that the username is valid (`@{hex_addr}(\..*)?:`) and that should do it for both cases

Reply:
I don't mean the test for the userid being valid, I meant integration tests.

I really don't see how this can be valuable, what is the benefit?

Reply:
This is exactly why I don't want to have to cases: https://github.com/raiden-network/raiden/pull/5173#issuecomment-548035373

Reply:
I'd argue we should remove the whole registration code (and the suffix) and only rely on the eth_auth. 
Why maintain and test code we're not using and aren't planning to use in the future.

Reply:
>  Why maintain and test code we're not using and aren't planning to use in the future.

This PR does remove one of the cases so that we only have to test one path (that is exactly why I don't want to add the non-prefix version back in, because of that one of the code paths was broken and untested). I'd prefer to keep the random userid as I can't see a way of enforcing/checking that the matrix operators are actually running the plugin and to me it seems unnecessary extra work having to curate the list of servers.